File: cpp/src/patterns/state/state/VendingMachineTest.cc
#include "VendingMachine.h"

int main () {
    VendingMachine *machine = new VendingMachine(5); 
    machine->Insert();
    machine->Bought();


    machine->Insert();
    machine->Eject();
    machine->Bought();

    machine->Insert();
    machine->Bought();
    machine->Insert();
    machine->Bought();
    machine->Eject();

    machine->Insert();
    machine->Insert();
    machine->Bought();
    machine->Insert();
    machine->Bought();
    machine->Insert();
    machine->Bought();
    return 0;
}

File: Java/src/patterns/state/naive/VendingMachineTest.java
package patterns.state.naive;

public class VendingMachineTest {

	public static void main(String[] args) {
		VendingMachine machine = new VendingMachine(5);
		machine.insert();
		machine.bought();


		machine.insert();
		machine.eject();
		machine.bought();

		machine.insert();
		machine.bought();
		machine.insert();
		machine.bought();
		machine.eject();

		machine.insert();
		machine.insert();
		machine.bought();
		machine.insert();
		machine.bought();
		machine.insert();
		machine.bought();
	}
}
